content/blog/2017-06-21-prometheus-20-alpha3-new-rule-format.md

```diff
@@ -0,0 +1,71 @@
+ 
+First, we moved to a new flag library, which uses the more common double-dash `--` prefix for flags instead of the single dash Prometheus used so far. Deployments have to be adapted accordingly.
+Additionally, further flags were removed with this alpha. The full list since Prometheus 1 is:
+ 
+`web.telemetry-path`<br/>
+All `storage.remote.*` flags<br/>
+All `storage.local.*` flags<br/>
+`query.staleness-delta`<br/>
+`alertmanager.url`<br/>
+ 
+Alerting and recording rules are on of the critical features of Prometheus.. But they also come with a few design issues and missing features, namely:
+ 
+* All rules are evaluated in concurrently, which is actually Prometheus’ oldest open bug. This has a couple of issues, the obvious one being that the load spikes every eval interval if you have a lot of rules. The other being that rules that depend on each other might not be fed the correct data. For example:
+
+```
+instance:network_bytes:rate1m = sum by(instance) (rate(network_bytes_total[1m]))
+
+ALERT HighNetworkTraffic
+  IF instance:network_bytes:rate1m > 10e6
+  FOR 5m
+```
+
+
+Here we are alerting over `instance:network_bytes:rate1m`, but `instance:network_bytes:rate1m` is itself being generated by another rule. We can get correct results only if the alert `HighNetworkTraffic` is run after the current value for `instance:network_bytes:rate1m` got recorded.
+
+
+* All rules run with the same interval. We could have some heavy rules that are better off run at 10min interval and some rules that could be run at 15sec intervals.
+* Rules and Alerts require the users to learn yet another DSL.
+ 
+To solve the issues above, grouping of rules has been proposed long back but has only recently been implemented as part of Prometheus 2.0. As part of this implementation we have also moved the rules to the YAML format as it doesn’t require a separate DSL for the users to pickup.
+ 
+Here’s how the new format looks:
+ 
+```yaml
+groups:
+- name: my-group-name
+  interval: 30s   # defaults to global interval
+  rules:
+  - record: errors:rate5m
+    expr: rate(errors_total[5m])
+  - record: requests:rate5m
+    Expr: rate(requests_total[5m])
+  - alert: HighErrors
+    # Expressions remain PromQL as before and can be spread over
+    # multiple lines via YAML’s multine strings.
+    expr: |
+      sum without(instance) (errors:rate5m)
+      / 
+      sum without(instance) (requests:rate5m)
+    for: 5m
+    labels:
+      severity: critical
+    annotations:
+      description: "stuff's happening with {{ $.labels.service }}"      
+```
+ 
+The rules in each group are executed sequentially and you can have an evaluation interval per group.
+ 
+As this change is breaking, we are going to release it with the 2.0 release and have added a command to promtool for the migration: `promtool update rules <filenames>`
+The converted files have the `.yaml` suffix appended and the `rule_files` clause in your Prometheus configuration has to be adapted.
```

Comment:
We use `prometheus.yml` as default. I guess we should only use one extension for YAML files.